From 4723efae70a4e5a3a2a10fc7441bd858ac4217cc Mon Sep 17 00:00:00 2001
From: Guobing Chen <guobing.chen@intel.com>
Date: Wed, 17 Apr 2019 16:12:38 +0800
Subject: [PATCH] modify makefile behavior to not remove pgo files when clean

The default 'make clean' will delete pgo files which leads pgo not
taking effect after make clean. Though that there is 'prof-clean'
available but that will not be used by autospec. Change the behavior of
'clean' to not remove pgo generated files, and add a 'clean-prof' target
to clean these files if needed.

Signed-off-by: Guobing Chen <guobing.chen@intel.com>
---
 Makefile.global | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile.global b/Makefile.global
index 3a7a18cc..6fea6f17 100644
--- a/Makefile.global
+++ b/Makefile.global
@@ -110,13 +110,15 @@ test: all
 	fi
 
 clean:
-	find . -name \*.gcno -o -name \*.gcda | xargs rm -f
 	find . -name \*.lo -o -name \*.o | xargs rm -f
 	find . -name \*.la -o -name \*.a | xargs rm -f
 	find . -name \*.so | xargs rm -f
 	find . -name .libs -a -type d|xargs rm -rf
 	rm -f libphp$(PHP_MAJOR_VERSION).la $(SAPI_CLI_PATH) $(SAPI_CGI_PATH) $(SAPI_LITESPEED_PATH) $(SAPI_FPM_PATH) $(OVERALL_TARGET) modules/* libs/*
 
+clean-prof:
+	find . -name \*.gcno -o -name \*.gcda | xargs rm -f
+
 distclean: clean
 	rm -f Makefile config.cache config.log config.status Makefile.objects Makefile.fragments libtool main/php_config.h main/internal_functions_cli.c main/internal_functions.c stamp-h buildmk.stamp Zend/zend_dtrace_gen.h Zend/zend_dtrace_gen.h.bak Zend/zend_config.h TSRM/tsrm_config.h
 	rm -f php7.spec main/build-defs.h scripts/phpize
@@ -143,5 +145,5 @@ prof-use:
 	CCACHE_DISABLE=1 $(MAKE) PROF_FLAGS=-fprofile-use all
 
 
-.PHONY: all clean install distclean test prof-gen prof-clean prof-use
+.PHONY: all clean install distclean test prof-gen prof-clean prof-use clean-prof
 .NOEXPORT:
-- 
2.20.1

